Gems/SceneProcessing/Code/Source/Generation/Components/TangentGenerator/TangentGenerateComponent.cpp

```diff
@@ -66,8 +70,37 @@ namespace AZ::SceneGenerationComponents
         return nullptr;
     }
 
+    void TangentGenerateComponent::GetRegistrySettings(
+        AZ::SceneAPI::DataTypes::TangentGenerationMethod& defaultGenerationMethod, bool& debugBitangentFlip) const
```

Comment:
Nit: I prefer returning a tuple to using output parameters. The main reason is that using output parameters prevents the values from being `const` at the call site. It is also not clear from the client code that the method accepts output parameters. You have to write:
```cpp
TangentGenerationMethod defaultGenerationMethod; // non-const
bool debugBitangentFlip; // non-const, may warn about being uninitialized
GetRegistrySettings(defaultGenerationMethod, debugBitangentFlip); // mutates the parameters
```

Using a tuple return type, you can instead write:
```cpp
const auto [defaultGenerationMethod, debugBitangentFlip] = GetRegistrySettings();
```

Reply:
Actually define a TangentGenerateRegSettings are better imo. This way you don't have to change the client code when adding more values.

Reply:
I'd want to know the places where those settings values are being read. If you're adding a new setting, presumably the place that reads the settings will need to be updated to account for that new setting.

Reply:
Value return semantics are nice, but definitely prefer a struct over a tuple. Tuples impact compile times fairly egregiously, and structs have nicer types usable in different contexts, and desugar in the same way.